### ISSUE TITLE
Wrong replace directive warning

### ISSUE THREAD

**Issue by ionik-dev** [ldez/gomoddirectives#5]
Hi, i am using your linter but met a bug(or couldn't understand linter logic)

There was a replace for one pkg, after work has been completed i removed line with replace(last line in go mod) executed `go mod tidy` and than got this error. But there is no line with replace in go.mod - how to fix this or is this a bug?
![image](https://user-images.githubusercontent.com/4531254/207769417-af4a5b64-1bee-4b07-b46e-53ae96ba931e.png)

**Comment by ldez** [comment]
Hi,

can you provide a minimal reproducible example?

**Comment by fgmarand** [comment]
Same problem here, and cannot provide a reproducible example as this happens with proprietary code. Like the original poster, in a recent commit there used to be a `replace` in the `go.mod` but it was removed and I still get the message. 

I suspect this is caused by a compilation cache somewhere. However `go clean -cache -testcache -fuzzcache -modcache` only made the check much slower, but did not prevent the error from reappearing. It looks like code from an other branch is being checked instead of the current one.

Here is the error message, though: the weird part being that it complains about `replace` then proceeds to show a `require` instead:

```
./bin/golangci-lint run --config ./build/.golangci.yml ./...
go.mod:7:1: replacement are not allowed: github.com/abhirockzz/dynamodb-local-testcontainers-go (gomoddirectives)
require (
^
```

And here are the first few lines of said `go.mod`. Line  7 is the `require`:

```go.mod
module github.com/<redacted>/<redacted>

go 1.22.0

toolchain go1.22.1

require (
	github.com/abhirockzz/dynamodb-local-testcontainers-go v0.0.0-20240209162301-43db52602e5c
	github.com/aws/aws-sdk-go-v2 v1.27.0
	github.com/aws/aws-sdk-go-v2/config v1.27.15
```

**Comment by mmb** [comment]
I am experiencing this too. I removed a replace directive but gomoddirectives is still saying the replacement is not allowed.

```
% golangci-lint run
go.mod:37:1: replacement are not allowed: github.com/enescakir/emoji (gomoddirectives)

^
1 issues:
* gomoddirectives: 1
% grep enescakir go.mod 
% 
```

The replace directive it's failing on was changed to a different replace in this commit: https://github.com/mmb/tmpbbs/commit/12d7cbf3309ca4d950386b0922c1d7d61356bf62

Something on my local development machine is causing this because in a fresh environment (GitHub Actions) it does not fail.
